tests/unit/test_config.py

```diff
@@ -0,0 +1,258 @@
+
+
+def test_config_validation(config_data):
+    """Test that config.json matches the expected schema and business rules."""
+    try:
+        validate_config(config_data)
+    except (ValueError, TypeError) as e:
+        pytest.fail(str(e))
+
+
+def test_field_types(config_data):
+    """Test that all fields have correct types."""
+    # Test string fields
+    string_fields = ["cubesat_name", "callsign", "super_secret_code", "repeat_code"]
+    for field in string_fields:
+        assert isinstance(config_data[field], str), f"{field} must be a string"
+
+    # Test numeric fields
+    float_fields = [
+        "last_battery_temp",
+        "NORMAL_CHARGE_CURRENT",
+        "NORMAL_BATTERY_VOLTAGE",
+        "CRITICAL_BATTERY_VOLTAGE",
+        "vlowbatt",
+        "battery_voltage",
+    ]
+    for field in float_fields:
+        assert isinstance(config_data[field], (int, float)), f"{field} must be a number"
+
+    int_fields = [
+        "sleep_duration",
+        "NORMAL_TEMP",
+        "NORMAL_BATT_TEMP",
+        "NORMAL_MICRO_TEMP",
+        "current_draw",
+        "REBOOT_TIME",
+    ]
+    for field in int_fields:
+        assert isinstance(config_data[field], int), f"{field} must be an integer"
+
+    # Test boolean fields
+    bool_fields = [
+        "detumble_enable_z",
+        "detumble_enable_x",
+        "detumble_enable_y",
+        "debug",
+        "legacy",
+        "heating",
+        "orpheus",
+        "is_licensed",
+        "turbo_clock",
+    ]
+    for field in bool_fields:
+        assert isinstance(config_data[field], bool), f"{field} must be a boolean"
+
+    # Test list fields
+    list_fields = ["jokes", "jokereply"]
+    for field in list_fields:
+        assert isinstance(config_data[field], list), f"{field} must be a list"
+        assert all(
+            isinstance(item, str) for item in config_data[field]
+        ), f"All items in {field} must be strings"
+
+    # Add radio config testing after list fields
+    assert isinstance(config_data["radio_cfg"], dict), "radio_cfg must be a dictionary"
+    radio_fields = {
+        "sender_id": int,
+        "receiver_id": int,
+        "transmit_frequency": float,
+        "LoRa_spreading_factor": int,
+        "transmit_bandwidth": int,
+        "LoRa_coding_rate": int,
+        "transmit_power": int,
+        "start_time": int,
+    }
+    for field, expected_type in radio_fields.items():
+        assert isinstance(
+            config_data["radio_cfg"][field], expected_type
+        ), f"radio_cfg.{field} must be a {expected_type.__name__}"
+
+
+def test_voltage_ranges(config_data):
+    """Test that voltage values are within expected ranges."""
+    voltage_fields = [
+        "battery_voltage",
+        "NORMAL_BATTERY_VOLTAGE",
+        "CRITICAL_BATTERY_VOLTAGE",
+        "vlowbatt",
+    ]
+    for field in voltage_fields:
+        value = config_data[field]
+        assert 0 <= value <= 12.0, f"{field} must be between 0V and 12V"
```

Comment:
You could actually constrain this further to be only within the acceptable range of our battery packs. 
```suggestion
        assert 5.2 <= value <= 8.4, f"{field} must be between 5.2V and 8.4V"
```